tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc

```diff
@@ -1666,49 +1754,65 @@ TEST_F(OpConverterTest, ConvertMatMul) {
         node_def, error::INVALID_ARGUMENT,
-        "transpose_a is not supported for TensorRT FullyConnected");
+        "Cannot transpose first input if it is a tensor with fewer than 2 "
+        "non-batch dimensions.");
   }
-  // OK.
-  for (bool transpose_b : {false, true}) {
+  {
+    // B must always have 2 non-batch dimensions
     Reset();
-    NodeDef node_def =
-        get_matmul_nodedef(DT_FLOAT, /*transpose_a=*/false, transpose_b);
+    NodeDef node_def = get_matmul_nodedef(DT_FLOAT, false, false);
     AddTestTensor("input", {2}, /*batch_size=*/1);
-    AddTestWeights<float>("weights", {2, 2}, {0, 1, 2, 3});
-    RunValidationAndConversion(node_def);
-    TRT_TensorOrWeights output;
-    TF_EXPECT_OK(GetTensorOrWeights("my_matmul", &output));
-    ASSERT_TRUE(output.is_tensor());
-    ExpectTrtDimsEqualsArray({2}, output.tensor()->getDimensions());
+    AddTestTensor("weights", {2}, /*batch_size=*/1);
+    RunValidationAndConversion(
+        node_def, error::INVALID_ARGUMENT,
+        "Second input must either be a constant, or contain at least 2 "
+        "non-batch dimensions.");
+  }
+  {
+    // We can never transpose weights that are not 2D.
+    Reset();
+    NodeDef node_def = get_matmul_nodedef(DT_FLOAT, true, false);
+    AddTestWeights<float>("input", {1, 1, 2}, {0, 1});
+    AddTestTensor("weights", {2, 2}, /*batch_size=*/1);
+    RunValidationAndConversion(
+        node_def, error::INVALID_ARGUMENT,
+        "Cannot currently transpose constant input if it is not 2 dimensional");
+  }
+  TestMatMulHelper(get_matmul_nodedef, "MatMul");
+}
 
-    const DataVec input_data{{"input", test::AsTensor<float>({0, 1})}};
-    DataVec output_data{{"my_matmul", ConstructTensor<float>(2)}};
-    BuildAndRun(input_data, &output_data);
-    if (transpose_b) {
-      EXPECT_THAT(GetSpanForData<float>(output_data[0]), ElementsAre(1, 3));
-    } else {
-      EXPECT_THAT(GetSpanForData<float>(output_data[0]), ElementsAre(2, 3));
-    }
+TEST_F(OpConverterTest, ConvertBatchMatMul) {
+  {
+    // Input list is empty, should fail.
+    NodeDef node_def = MakeNodeDef("my_matmul", "BatchMatMul", {});
+    RunValidationAndConversion(
+        node_def, error::INVALID_ARGUMENT,
+        "BatchMatMul got 0 inputs but expected 2, at my_matmul");
   }
+
+  // Get the NodeDef for BatchMatMul.
+  auto get_matmul_nodedef = [](DataType dtype, bool transpose_a,
+                               bool transpose_b) -> NodeDef {
+    Scope s = Scope::NewRootScope();
+    auto input = ops::Placeholder(s.WithOpName("input"), dtype);
+    auto weights = ops::Placeholder(s.WithOpName("weights"), dtype);
+    const auto matmul_attrs =
+        ops::BatchMatMul::AdjX(transpose_a).AdjY(transpose_b);
+    auto matmul = ops::BatchMatMul(s.WithOpName("my_matmul"), input, weights,
+                                   matmul_attrs);
+    return matmul.operation.node()->def();
+  };
+
+  TestMatMulHelper(get_matmul_nodedef, "BatchMatMul");
```

Comment:
As mentioned above we don't have a positive case for batchmatmul?

Reply:
I've added a positive test case